pkg/skaffold/kubernetes/status/status_check.go

```diff
@@ -220,17 +221,19 @@ func (s *monitor) statusCheck(ctx context.Context, out io.Writer) (proto.StatusC
 
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
+	var exitStatusCode proto.StatusCode
 
 	for _, d := range resources {
 		wg.Add(1)
 		go func(r *resource.Resource) {
 			defer wg.Done()
 			// keep updating the resource status until it fails/succeeds/times out
 			pollResourceStatus(ctx, s.cfg, r)
-			rcCopy := c.markProcessed(r.Status().Error())
+			rcCopy := c.markProcessed(ctx, r.StatusCode())
```

Comment:
Can we move the logic here and in `getSkaffoldDeployStatus()` into `counter`?  This scattering of logic is awkward:
- `markProcessed` can check if `ctx.Err() == context.Canceled` to determine if the user cancelled
- have `markProcessed` return `(counter, failed bool)` to replace the `resourceFailed()` check on line 235
- move the exitStatusCode determination into `counter`
- line 249 would `return c.result()`

Reply:
We can't use `markProcessed can check if ctx.Err() == context.Canceled to determine if the user cancelled` since, we fail fast and cancel context if a deployment failed.
So at this point, its difficult to detect if a user hit cntrl C Vs, code cancelling all the status check for all resources.

Reply:
I had suggestions 2-4 implemented but then the unit tests were becoming more complex. Maybe the `counter` need to be renamed to `deployState`.

Reply:
Ok, so i remember, why we cant have counter store the `exitStatusCode`. The counter gets updated in a go routine. 
The exit code is stored in `resources`. It makes sense to declare another mutex to update the exit status code safely.

Reply:
Ok, 
I am addressed 1 and 2 from the suggestions. 
For 3, 4: 
 - I have declared a exitStatusCode in `checkStatus` and updated the code to only capture first failure.
 -  kept the `getDeployStatus` as is.